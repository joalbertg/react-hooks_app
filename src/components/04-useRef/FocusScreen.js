import React, { useRef } from 'react';

import './focus.css';

export const FocusScreen = () => {
  const inputRef = useRef();
  const handleClick = () => {
    //document.querySelector('input').select();
    //console.log(inputRef)
    inputRef.current.select();
  }

  return(
    <>
      <h1>FocusScreen</h1>
      <hr />

      <input
        ref={inputRef}
        className="form-control"
        type="text"
        placeholder="Your name"
      />
      <button
        className="btn btn-info mt-2"
        onClick={handleClick}
      >
        Focus
      </button>
    </>
  );
}

