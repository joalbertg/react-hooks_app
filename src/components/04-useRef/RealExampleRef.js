import React, { useState } from 'react';

import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import './focus.css';

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);
  const handleClick = () => setShow(!show);

  return(
    <>
      <h1>RealExampleRef</h1>
      <hr />

      { show && <MultipleCustomHooks /> }
      <button
        className="btn btn-info mt-2"
        onClick={handleClick}
      >
        Show/Hide
      </button>
    </>
  );
}

