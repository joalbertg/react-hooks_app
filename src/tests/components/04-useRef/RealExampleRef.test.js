import React from 'react';
import { shallow } from 'enzyme';
import { renderHook, act } from '@testing-library/react-hooks';

import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Tests RealExampleRef component', () => {
  const wrapper = shallow(<RealExampleRef />);

  test('Should display correctly', () => {
    const MultipleCustomHooks = wrapper.find('MultipleCustomHooks');

    expect(wrapper).toMatchSnapshot();
    expect(MultipleCustomHooks.exists()).toBe(false);
  });

  test('Should display MultipleCustomHooks component', () => {
    wrapper.find('button').simulate('click');

    const MultipleCustomHooks = wrapper.find('MultipleCustomHooks');

    expect(wrapper).toMatchSnapshot();
    expect(MultipleCustomHooks.exists()).toBe(true);
  });
});

