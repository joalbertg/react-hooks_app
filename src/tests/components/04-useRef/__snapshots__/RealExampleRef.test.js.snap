// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests RealExampleRef component Should display MultipleCustomHooks component 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-info mt-2"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;

exports[`Tests RealExampleRef component Should display correctly 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-info mt-2"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
